controller/src/main/java/io/pravega/controller/store/task/InMemoryTaskMetadataStore.java

```diff
@@ -32,13 +30,9 @@
     @GuardedBy("itself")
     private final Map<Resource, LockData> lockTable;
 
-    @GuardedBy("itself")
-    private final Map<String, Set<TaggedResource>> hostTable;
-
     InMemoryTaskMetadataStore(ScheduledExecutorService executor) {
-        super(executor);
+        super(new InMemoryHostIndex(), executor);
         lockTable = new HashMap<>();
```

Comment:
Some methods perform CAS on keys in the map, hence locking may not be avoidable even with ConcurrentHashMap.